### PR TITLE
@W-13037169 | Add Sample CCA Orchestrator Utest

### PR DESCRIPTION
**Pull request by jbisaSF**
**Overview**

Adding a sample orchestrator utest class and required dependencies to unit test a custom CCA orchestrator.

**Testing Details**

- Ran these test cases in an org in Steam and seeing `78%` code coverage for the sample orchestrator.

**Testing Screenshots**

Screenshot 2023-08-01 at 12 40 18 PM

### CODE REVIEW

File: commerce/domain/orchestrators/classes/CartCalculateExecutorForUnitTest.cls [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/**
+* @description Sample mock executor for unit testing the various calculators within Cart Calculate API
```
Comment (jbisaSF) [inline commerce/domain/orchestrators/classes/CartCalculateExecutorForUnitTest.cls:2]: @pplat23 should we go with "for unit testing a custom orchestrator within Cart Calculate API" ?

File: commerce/domain/orchestrators/classes/CartCalculateExecutorMock.cls [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+* @description Mock executor for the various calculators within Cart Calculate API. Should be used only for tests.
+*/
+global abstract class CartCalculateExecutorMock {
```
Comment (pplat23) [inline commerce/domain/orchestrators/classes/CartCalculateExecutorMock.cls:4]: Shouldn't this class be a file based Apex and reside in core codebase?

File: commerce/domain/orchestrators/classes/CartCalculateExecutorForUnitTest.cls [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+     * @description All classes extending CartExtension.CartCalculateExecutorMock must have a default constructor defined
+     */
+    global CartCalculateExecutorForUnitTest() {}
+
+    global override void prices(CartExtension.CartCalculateCalculatorRequest request) {
+        CartExtension.Cart cart = request.getCart();
+        cart.setName('CartRepriced');
+    }
+
+    global override void promotions(CartExtension.CartCalculateCalculatorRequest request) {}
```
Comment (pplat23) [inline commerce/domain/orchestrators/classes/CartCalculateExecutorForUnitTest.cls:16]: why don't we implement all this methods in the Sample?
`cart.setName(cart.getName() + 'PromotionsRecalculated');`

File: commerce/domain/orchestrators/classes/CartCalculateSampleUnitTest.cls [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/**
+ * A Sample unit test for <<CartCalculate>>.
+ */
+@IsTest
+public class CartCalculateSampleUnitTest {
+
+    @IsTest
+    public static void testWhenCartCalculateIsCalledAtStartCheckoutThenNoCVOIsCreated() {
```
Comment (pplat23) [inline commerce/domain/orchestrators/classes/CartCalculateSampleUnitTest.cls:8]: `ThenPricingIsExecuted`?

**Review by smahbubani99 — APPROVED** [review]

**Pull request merged**